src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,182 @@
+// Solution below inspired by ChatGPT, to solve the issue in the else block of incrementing the num_items counter to add
+// the new item subsequently.
+/* Solution below inspired by ChatGPT, to solve the issue for the command list, to show the separators only at the start
+and end, by moving
+the statement outside the for loop
+ */
+// Solution below inspired by ChatGPT, regarding the method .contains("mark"), as it might be a bug if the input
+// contains mark, without a corresponding integer.
+// Solution below inspired by https://stackoverflow.com/questions/12973871/read-multiple-user-input-words-and-split-them
+// Solution below inspired by ChatGPT, employed its help to solve my indexing problem. (ie mark 1 and unmark 1 refers to
+// diff items)
+/* Solution below inspired by ChatGPT, employed its help to solve the list not updating and showing the items with their
+correct status icon, by creating a new task array of tasks instead of a string array.
+ */
+// Solution below inspired by https://stackoverflow.com/questions/10405789/regex-append-or-replace-only-the-first-
+// letter-of-each-word
+// Solution below inspired by https://www.programiz.com/java-programming/library/string/replacefirst
+// Solution below adapted by ChatGPT, to solve the exception error when invoking last line of the loop.
+// when there is no next line.
+// Solution below inspired by https://stackoverflow.com/questions/32733084/pass-a-simple-enum-into-a-constructor-in-java
+// Solution below inspired from ChatGPT, seeked clarification if the enums have to be passed into the child classes of
+// parent class Task's constructor
+
+/**
+ * The main class for the Duke application.
+ * Duke is a task management application that allows users to manage their tasks, including todos, deadlines, and events
+ */
+public class Duke {
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private static String filePath = "./data/duke.txt";
+    private Stage stage;
+    private static Ui ui;
+    private static Storage storage;
+    private static TaskList taskList;
+    private static Parser parser;
+
+    public Duke() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        ui = new Ui();
+        storage = new Storage(filePath);
+        taskList = new TaskList();
+        parser = new Parser();
+
+        try {
+            tasks = storage.loadTasks(); // Load tasks from file
+            taskList.setTasks(tasks);
+        } catch (IOException e) {
+            System.out.println("Error loading tasks: " + e.getMessage());
+        }
+        ui.displayWelcomeText();
+    }
+
+    String getResponse(String input) {
+        try {
+            Command command = parser.parseCommand(input);
+            return command.execute(taskList, ui, storage);
+        } catch (DukeException.ToDoException e) {
+            return ui.printToDoException();
```

Comment:
Too many duplicate code here.

Consider doing something like
`catch (DukeException.ToDoException e) {
      return ui.printException(e.getMessage())
 }`
 and store the message directly in your exception?